Documentation/Books/Manual/Upgrading/VersionSpecific/Upgrading34.md

```diff
@@ -1,6 +1,29 @@
 Upgrading to ArangoDB 3.4
 =========================
 
-Please be sure that you have checked the list of [incompatible changes in 3.4](../../ReleaseNotes/UpgradingChanges34.md)
+Please be sure that you have checked the list of
+[incompatible changes in 3.4](../../ReleaseNotes/UpgradingChanges34.md)
 before upgrading.
 
+Upon upgrading from 3.3 to 3.4, the following storage engine-specific data conversion tasks
+will be executed:
+
+- **MMFiles storage engine:**
+
+  All collection datafiles will be rewritten into a
+  new data format. This data format is required in order to support using the collections
+  with the ArangoSearch tool introduced in ArangoDB 3.4. 
+
+  The conversion will read each datafile sequentially and write out a new datafile in the
+  new format sequentially. This means the disk will be involved, but both reading and
+  writing are done in a sequential fashion. Preliminary tests have shown that it will need 
```

Comment:
So even with multiple collections and databases, all datafiles are processed one after another (single-threaded)?